### ISSUE TITLE
Polygon drawing in Labels layer should use a consistent UI

### ISSUE THREAD

**Issue by GenevieveBuckley** [napari/napari#6259]
## 🚀 Feature
The polygon function in the Labels layer and the polygon function in the Shapes layer should both have the same type of user interaction controls.

## Motivation

The motivation for this is to provide a consistent UI across all of napari. It's frustrating for users to interact with layer controls work one way for some layers, and in a totally different way for other layers.

## Pitch

I want the polygon function in the Labels layer to "complete" the shape when the user double-clicks anywhere on the canvas (i.e. have the same UI as drawing point-by-point polygons in the shapes layer). I'd also like the documentation updated to match.

## Alternatives

The current implementation restricts polygon shape completion to when the user double clicks within 20 pixels of the first polygon point made. This is not ideal because (1) it is not consistent UI with the polygons in the Shapes layer, (2) for large images with many pixels it can be hard to click precisely enough to fit within the 20 pixel distance without zooming in a lot, and (3) I've sometimes forgotten which was the first polygon point and messed up the shape at the very end.

## Additional context

Original feature PR: https://github.com/napari/napari/pull/5806
Documentation PR: https://github.com/napari/docs/pull/177

**Comment by psobolewskiPhD** [comment]
I agree--the double-click to end came up in review of the original PR. There was some discussion as to whether it was a standard function or not. 
cc: @jni and @melonora who were also involved.
Ultimately I think it got shelved due to a linux issue with double-click speed, see:
https://github.com/napari/napari/pull/5806#issuecomment-1567419329
(discussion starts few posts higher)

**Comment by GenevieveBuckley** [comment]
I see, thank you. Apparently this was not a deal breaker though, if we see earlier in the thread: https://github.com/napari/napari/pull/5806#issuecomment-1567447598

My objection is to the restriction that this happens only within a certain radius of the original polygon starting point. That makes it difficult to use in practice - there's more mouse movement, because you effectively now have to do an extra point per shape, it's hard to be that accurate with extremely high resolution images when you are zoomed out to see the whole view, etc.

**Comment by jni** [comment]
> it's hard to be that accurate with extremely high resolution images when you are zoomed out to see the whole view, etc.

I thought the minimum distance was in screen pixels? It's also configurable in the API, if not yet the settings, in the LabelsPolygonOverlay.completion_radius value. Set that to a high value and you get double click anywhere. It is a goal to allow users to change that in the settings.

**Comment by GenevieveBuckley** [comment]
> It is a goal to allow users to change that in the settings.

This is really not the point, the problem is that you use a different UI for the same task in two different places.

If the consensus is that method A is better than method B and should be used everywhere, that's fine, even if method A is not my preference. But we can't handle the UI differently for the same user task.

**Comment by jni** [comment]
@GenevieveBuckley yes, that's definitely the plan. If you look at the discussion in #5806, @melonora, @psobolewskiPhD and I come to the consensus that eventually, we want to have the idea of *raster data* and *vector data*, and unified editing tools between the two. But your pitch above advocates for consistency in a specific direction. 😉 

As I think more about unification, though, some things are easy or trivial for vector data and extremely hard or impossible for raster data: vertex editing, rotating, overlapping shapes, ... Also, converting from one shape class to another is lossy. So I think a complete unification is not achievable. However, the basic "painting" tools should be unified.

**Comment by psobolewskiPhD** [comment]
This is closed by:
https://github.com/napari/napari/pull/7023
and
https://github.com/napari/napari/pull/7019

There a few differences that are still noted in this more general issue:
https://github.com/napari/napari/issues/6489
